sdcm/nemesis.py

```diff
@@ -445,18 +445,79 @@ def disrupt_nodetool_cleanup(self):
         cmd = 'nodetool -h localhost cleanup keyspace1'
         self._run_nodetool(cmd, self.target_node)
 
+    def _run_in_cqlsh(self, cmd, node=None):
+        if not node:
+            node = self.target_node
+        node.remoter.run('cqlsh -e "{}" {}'.format(cmd, node.private_ip_address), verbose=True)
+
+    def _modify_table_property(self, name, val, keyspace="keyspace1", table="standard1"):
+        self._set_current_disruption('ModifyTableProperties%s %s' % (name.capitalize(), self.target_node))
+        cmd = "ALTER TABLE {keyspace}.{table} WITH {name} = {val};".format(**locals())
+        self._run_in_cqlsh(cmd)
+
     def disrupt_modify_table_comment(self):
-        self._set_current_disruption('ModifyTableProperties %s' % self.target_node)
-        comment = ''.join(random.choice(string.ascii_letters) for i in xrange(24))
-        cmd = "ALTER TABLE keyspace1.standard1 with comment = '{}';".format(comment)
-        self.target_node.remoter.run('cqlsh -e "{}" {}'.format(cmd, self.target_node.private_ip_address), verbose=True)
+        prop_val = ''.join(random.choice(string.ascii_letters) for _ in xrange(24))
+        self._modify_table_property(name="comment", val="'%s'" % prop_val)
 
     def disrupt_modify_table_gc_grace_time(self):
-        self._set_current_disruption('ModifyTableProperties %s' % self.target_node)
-        gc_grace_seconds = random.choice(xrange(216000, 864000))
-        cmd = "ALTER TABLE keyspace1.standard1 with comment = 'gc_grace_seconds changed' AND" \
-              " gc_grace_seconds = {};".format(gc_grace_seconds)
-        self.target_node.remoter.run('cqlsh -e "{}" {}'.format(cmd, self.target_node.private_ip_address), verbose=True)
+        self._modify_table_property(name="gc_grace_seconds", val=random.randint(216000, 864000))
+
+    def disrupt_modify_table_caching(self):
+        prop_val = dict(
+            keys=random.choice(["NONE", "ALL"]),
+            rows_per_partition=random.choice(["NONE", "ALL", random.randint(1, 10000)])
+        )
+        self._modify_table_property(name="caching", val=str(prop_val))
```

Comment:
Can you please add in comments, what is the default value it starts with and maybe part of documentation that explain how it should affect schema behaviour?

Reply:
Will do :+1: